benchmarks/src/random_integer_generation.cpp

```diff
@@ -0,0 +1,98 @@
+#include <benchmark/benchmark.h>
+#include <random>
+
+/// Test URBGs alone
+
+static void BM_mt19937(benchmark::State& state) {
+    std::mt19937 gen;
+    for (auto _ : state) {
+        benchmark::DoNotOptimize(gen());
+    }
+}
+BENCHMARK(BM_mt19937);
+
+static void BM_mt19937_64(benchmark::State& state) {
+    std::mt19937_64 gen;
+    for (auto _ : state) {
+        benchmark::DoNotOptimize(gen());
+    }
+}
+BENCHMARK(BM_mt19937_64);
+
+static void BM_lcg(benchmark::State& state) {
+    std::minstd_rand gen;
+    for (auto _ : state) {
+        benchmark::DoNotOptimize(gen());
+    }
+}
+BENCHMARK(BM_lcg);
+
+uint32_t GetMax() {

```

Comment:
Technically, we should include `<cstdint>` and `std`-qualify these types. Occurs below.